pkg/controller/attach_detach.go

```diff
@@ -191,11 +192,22 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 				response, attachErr = r.attachVolume(cloudCtx, azVolumeAttachment.Spec.VolumeID, azVolumeAttachment.Spec.NodeName, azVolumeAttachment.Spec.VolumeContext)
 			}
 		}
+		pods, err := r.controllerSharedState.getPodsFromVolume(ctx, r.client, azVolumeAttachment.Spec.VolumeName)
+		if err != nil {
+			klog.Infof("failed to get pods for volume (%s). Error: %v", azVolumeAttachment.Spec.VolumeName, err)
+		}
+
 		// update AzVolumeAttachment CRI with the result of the attach operation
 		var updateFunc func(interface{}) error
 		if attachErr != nil {
 			klog.Errorf("failed to attach volume %s to node %s: %v", azVolumeAttachment.Spec.VolumeName, azVolumeAttachment.Spec.NodeName, attachErr)
 
+			if len(pods) > 0 {
+				for _, pod := range pods {
+					r.controllerSharedState.eventRecorder.Eventf(pod.DeepCopyObject(), v1.EventTypeWarning, consts.ReplicaAttachmentFailedEvent, "%s mount for volume %s could not attached to node %s", azVolumeAttachment.Spec.RequestedRole, azVolumeAttachment.Spec.VolumeName, azVolumeAttachment.Spec.NodeName)
```

Comment:
It would be useful to include the errors message in the event as well.

Reply:
I'm not sure if it would be helpful for the customer to understand the error message. I think the error message would be helpful for someone who's debugging the failure but in that case the message is being written in the logs anyway.

One scenario when the customer can do something about the error would be to increase the number of nodes in case there aren't enough suitable nodes and we have a separate error message for that. 

Do you think there can be other scenarios where the customer can take action based on the error message?

Reply:
While I agree that the customer may not be able to take action on the error message, it's likely that event will be part of any CRI raised by the customer in relation to the error. If the error message is in the event, it would help to jump start any investigation on our part /wo having to ask the customer to produce the relevant logs.